File: JAVA/Lista/PrincipalLista.java
package JAVA.Lista;

public class PrincipalLista {
    public static void main(String[] args) {
        Lista lista = new Lista();
        lista.agregar("Argentina");
        lista.agregar("Peru");
        lista.agregar("Brasil");
        lista.agregar("Chile");
        lista.agregar("Uruguay");
        lista.agregar("Colombia");

        System.out.println("********************");
        System.out.println("Buscando 'Brazil':");
        System.out.println("********************");
        if (lista.buscar("Brazil") != null) {
            System.out.println("Encontrado");
        } else {
            System.out.println("No encontrado");
        }

        System.out.println("********************");
        System.out.println("Buscando 'Brasil':");
        System.out.println("********************");
        if (lista.buscar("Brasil") != null) {
            System.out.println("Encontrado");
        } else {
            System.out.println("No encontrado");
        }

        System.out.println("********************");
        System.out.println("Lista original: ");
        System.out.println("********************");
        lista.imprimir();

        System.out.println("********************");
        System.out.println("\nLista ordenada: ");
        System.out.println("********************");
        lista.ordenar();
        lista.imprimir();

        System.out.println("********************");
        System.out.println("\nLista desordenada: ");
        System.out.println("********************");
        lista.desordenar();
        lista.imprimir();

        System.out.println("*************************");
        System.out.println("\nPais eliminado 'Chile'");
        lista.eliminar("Chile");
        System.out.println("*************************");
        lista.imprimir();
        

        System.out.println("******************************");
        System.out.println("\nPais insertado 'Venezuela'");
        lista.insertar(1, "Venezuela");
        System.out.println("******************************");
        lista.imprimir();
        lista.ordenar();

        System.out.println("******************************");
        System.out.println("\nLista recorrido inicio-fin: ");
        lista.recorridoInicioFin();
        System.out.println("******************************");

        System.out.println("******************************");
        System.out.println("\nLista recorrido fin-inicio: ");
        lista.recorridoFinInicio();
        System.out.println("******************************");
    }
}

File: PYTHON/Lista/PrincipalLista.py
#Implement all the methods from the Lista class
from Lista import Lista

lista = Lista()

lista.agregar("Argentina")
lista.agregar("Bolivia")
lista.agregar("Ecuador")
lista.agregar("Paraguay")
lista.agregar("Chile")
lista.agregar("México")
lista.agregar("Espana")

print("********************")
print("Buscando 'Brazil':")
print("********************")
if lista.buscar("Brazil"):
    print("El pais esta en la lista")
else:
    print("El pais NO esta en la lista")

print("********************")
print("Buscando 'Brasil':")
print("********************")
if lista.buscar("Brasil"):
    print("El pais esta en la lista")
else:
    print("El pais NO esta en la lista")

print("********************")
print("Lista original: ")
print("********************")
lista.imprimir()

print("********************")
print("Lista ordenada: ")
print("********************")
lista.ordenar()
lista.imprimir()

print("********************")
print("Lista desordenada: ")
lista.desordenar()
print("********************")
lista.imprimir()

print("********************")
print("Pais eliminado 'Chile'")
lista.eliminar("Chile")
print("********************")
lista.imprimir()

print("********************")
print("Pais insertado 'Perú' en la posicion 2")
lista.insertar(2,"Perú")
print("********************")
lista.imprimir()

print("*******************************")
print("Lista recorrido inicio a fin: ")
print("*******************************")
lista.recorridoInicioFin()

print("*******************************")
print("Lista recorrido fin a inicio: ")
print("*******************************")
lista.recorridoFinInicio()